test/e2e/gameserver_test.go

```diff
@@ -907,6 +907,107 @@ func TestGameServerTcpUdpProtocol(t *testing.T) {
 	assert.Equal(t, "ACK TCP: Hello World !\n", replyTCP)
 }
 
+// TestGameServerStaticTcpUdpProtocol checks UDP and TCP connection for TCPUDP Protocol of Static Portpolicy
+func TestGameServerStaticTcpUdpProtocol(t *testing.T) {
+	framework.SkipOnCloudProduct(t, "gke-autopilot", "does not support Static PortPolicy")
+	t.Parallel()
+	gs := framework.DefaultGameServer(framework.Namespace)
+	gs.Spec.Ports[0].PortPolicy = agonesv1.Static
+	gs.Spec.Ports[0].Protocol = agonesv1.ProtocolTCPUDP
+	gs.Spec.Ports[0].HostPort = 7000
+	gs.Spec.Ports[0].Name = "gameserver"
+	gs.Spec.Template.Spec.Containers[0].Env = []corev1.EnvVar{{Name: "TCP", Value: "TRUE"}}
+
+	errs := gs.Validate(agtesting.FakeAPIHooks{})
+	require.Len(t, errs, 0)
+
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(t, framework.Namespace, gs)
+	if err != nil {
+		assert.FailNow(t, "Could not get a GameServer ready", err.Error())
+	}
```

Comment:
Nit:

```suggestion
  require.NoError(t, err)
```

Same for below.

Reply:
Bumping this final nit (and below), then this is good to merge.

Reply:
I think I accidentally resolved this yesterday when I didn't mean to.

if we can convert all the 

```
	if err != nil {
		assert.FailNow(t, ...)
	}
```

Statements into `require.NoError(t, err)` that would be the last thing needed before this is good to merge.